Uploader.m

```diff
@@ -50,104 +85,119 @@ - (void)uploadFiles:(RNFSUploadParams*)params
-  } else {
-    sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
-  }
     
-  NSURLSession *session = [NSURLSession sessionWithConfiguration:sessionConfiguration delegate:(id)self delegateQueue:[NSOperationQueue mainQueue]];
-  _task = [session dataTaskWithRequest:req completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-      NSString * str = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
-      return _params.completeCallback(str, response);
-  }];
-  [_task resume];
-  _params.beginCallback();
+  return nil;
 }
 
-- (NSString *)generateBoundaryString
+- (NSURLSessionDataTask *)createUploadTaskFromReq: (NSMutableURLRequest *)req
 {
-  NSString *uuid = [[NSUUID UUID] UUIDString];
-  return [NSString stringWithFormat:@"----%@", uuid];
+    NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:sessionConfiguration delegate:(id)self delegateQueue:[NSOperationQueue mainQueue]];
+    return [session dataTaskWithRequest:req completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+      NSLog(@"---INLINE COMPLETED EMIT---");
+      NSString * str = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+      return _params.completeCallback(str, response);
+    }];
 }
 
-- (NSString *)mimeTypeForPath:(NSString *)filepath
+- (NSURLSessionDataTask *)createBackgroundUploadTaskFromReq: (NSMutableURLRequest *)req fromFilePath:(NSString *)filepath
 {
-  NSString *fileExtension = [filepath pathExtension];
-  NSString *UTI = (__bridge_transfer NSString *)UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, (__bridge CFStringRef)fileExtension, NULL);
-  NSString *contentType = (__bridge_transfer NSString *)UTTypeCopyPreferredTagWithClass((__bridge CFStringRef)UTI, kUTTagClassMIMEType);
-
-  if (contentType) {
-    return contentType;
-  }
-  return @"application/octet-stream";
+    NSURL *fileUrl = [NSURL fileURLWithPath:filepath];
+    NSString *uuid = [[NSUUID UUID] UUIDString];
+    NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:uuid];
+    sessionConfiguration.sessionSendsLaunchEvents = YES;
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:sessionConfiguration delegate:(id)self delegateQueue:[NSOperationQueue mainQueue]];
+    return [session uploadTaskWithRequest:req fromFile:fileUrl];
 }
 
 - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
 {
+  NSLog(@"---COMPLETED EMIT---");
   if(error != nil) {
     return _params.errorCallback(error);
   }
+    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)task.response;
+    NSString *body = @"rock your body";
```

Comment:
But I can turn that into a todo...there's a different lifecycle event we need to handle to recieve body data from the response...